distributed/scheduler.py

```diff
@@ -3035,6 +3038,18 @@ def get_task_stream(self, comm=None, start=None, stop=None, count=None):
         ts = [p for p in self.plugins if isinstance(p, TaskStreamPlugin)][0]
         return ts.collect(start=start, stop=stop, count=count)
 
+    @gen.coroutine
+    def register_worker_callbacks(self, comm, setup=None):
+        """ Registers a setup function, and call it on every worker """
+        if setup is None:
+            raise gen.Return({})
+
+        self.worker_setups.append(setup)
+
+        responses = yield self.broadcast(msg=dict(op='run', function=setup,
+                                                  args=(), kwargs={}))
```

Comment:
Not sure why, but if I don't force `args=(), kwargs={}` here, and don't use `*args, **kwargs` signature for the `register_worker_callbacks` to propagate them as before, I got the following exception when running `register_worker_callbacks`:

````
Traceback (most recent call last):
  File "/home/eh/eynardbg/git/distributed/distributed/core.py", line 340, in handle_comm
    result = yield result
  File "/work/ADM/hpc/eynardbg/Outils/miniconda3/envs/dask_dev/lib/python3.5/site-packages/tornado/gen.py", line 1133, in run
    value = future.result()
  File "/work/ADM/hpc/eynardbg/Outils/miniconda3/envs/dask_dev/lib/python3.5/asyncio/futures.py", line 294, in result
    raise self._exception
  File "/work/ADM/hpc/eynardbg/Outils/miniconda3/envs/dask_dev/lib/python3.5/site-packages/tornado/gen.py", line 326, in wrapper
    yielded = next(result)
  File "/home/eh/eynardbg/git/distributed/distributed/worker.py", line 1036, in run
    kwargs = pickle.loads(kwargs)
  File "/home/eh/eynardbg/git/distributed/distributed/protocol/pickle.py", line 61, in loads
    logger.info("Failed to deserialize %s", x[:10000], exc_info=True)
TypeError: unhashable type: 'slice'
````

Investigating, it seems like a kwargs contains a dask-worker key and is a plain non serialized dict. Really don't know how it gets passed to `worker.run()` method. That was the problem I tried to describe [here](https://github.com/dask/distributed/pull/2201#discussion_r213552607).